plugin-viewport/src/test/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImplTest.kt

```diff
@@ -0,0 +1,444 @@
+package com.mapbox.maps.plugin.viewport
+
+import android.os.Handler
+import com.mapbox.geojson.Point
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.plugin.animation.CameraAnimationsPlugin
+import com.mapbox.maps.plugin.animation.Cancelable
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.delegates.MapPluginProviderDelegate
+import com.mapbox.maps.plugin.locationcomponent.LocationComponentPlugin
+import com.mapbox.maps.plugin.locationcomponent.location
+import com.mapbox.maps.plugin.viewport.data.DefaultViewportTransitionOptions
+import com.mapbox.maps.plugin.viewport.data.FollowingViewportStateOptions
+import com.mapbox.maps.plugin.viewport.data.OverviewViewportStateOptions
+import com.mapbox.maps.plugin.viewport.state.ViewportState
+import com.mapbox.maps.plugin.viewport.transition.ViewportTransition
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.slot
+import io.mockk.unmockkAll
+import io.mockk.unmockkStatic
+import io.mockk.verify
+import io.mockk.verifyOrder
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+@MapboxExperimental
+class ViewportPluginImplTest {
+  private val delegateProvider = mockk<MapDelegateProvider>(relaxed = true)
+  private val mapPluginProviderDelegate = mockk<MapPluginProviderDelegate>(relaxed = true)
+  private val cameraPlugin = mockk<CameraAnimationsPlugin>(relaxed = true)
+  private val locationPlugin = mockk<LocationComponentPlugin>(relaxed = true)
```

Comment:
I understand we're a bit short on time but using fully relaxed mocks (and not `relaxedUnitFun`) may hide some issues. Perhaps we could mock a bit more if it's not a big burden and relax only unit functions

Reply:
Yeah, it's mainly used to speed up writing these tests to ship in beta, will revisit tests after the beta release.

Reply:
resolved in https://github.com/mapbox/mapbox-maps-android/pull/1074